src/app/beer_garden/db/mongo/models.py

```diff
@@ -886,6 +887,50 @@ class User(Document):
         "indexes": [{"name": "unique_index", "fields": ["username"], "unique": True}],
     }
 
+    _permissions_cache: Optional[dict] = None
+
+    @property
+    def permissions(self) -> dict:
```

Comment:
Hopefully between the docstrings and the actual usage it's clear why I added this permissions caching related stuff to the User model.  Do ask questions or provides suggestions though if it's not so clear.

Reply:
This looks fine. Different from the other caching property, this doesn't rely on a property access as the sole method to manipulate the internal state of an object, unlike what the framework forced you to do above.